src/main/java/io/akka/monitoring/application/IoTDeviceTemperatureStream.java

```diff
@@ -0,0 +1,105 @@
+package io.akka.monitoring.application;
+
+import akka.Done;
+import akka.NotUsed;
+import akka.javasdk.client.ComponentClient;
+import akka.stream.javadsl.Source;
+import io.akka.monitoring.domain.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Simulates a stream of temperature measurements from IoT devices located in different rooms.
+ * It feeds the system with both historical data (last 10 minutes) and real-time measurements.
+ *
+ * In real-world applications, this fake steam would be replaced with a {@link akka.javasdk.consumer.Consumer}
+ * that reads from a real-time data source.
+ */
+public class IoTDeviceTemperatureStream {
+
+  private static final Logger log = LoggerFactory.getLogger(IoTDeviceTemperatureStream.class);
+  private final ComponentClient componentClient;
+  private final Random random = new Random();
+  public final List<Location> LOCATIONS = List.of(
+    new Location("temp-101", "Boiler Room A"),
+    new Location("temp-102", "Server Room B"),
+    new Location("temp-103", "Warehouse C"));
+
+  public IoTDeviceTemperatureStream(ComponentClient componentClient) {
+    this.componentClient = componentClient;
+  }
+
+  record RawTemperatureMeasurement(
+    Location location,
+    double temperature,
+    Instant timestamp
+  ) {
+    public String aggregationId() {
+      return timestamp.truncatedTo(ChronoUnit.MINUTES).toString() + AggregatedTemperature.SEPARATOR + location.sensorId();
+    }
+
+    public RawTemperatureMeasurement withTimestamp(Instant newTimestamp) {
+      return new RawTemperatureMeasurement(location, temperature, newTimestamp);
+    }
+  }
+
+  public Source<Done, NotUsed> createStream() {
+
+    return historicalDataSource().concat(Source.tick(
```

Comment:
I don't think this matter at all here, in this example. Since the data is minimal. Just noting that with more data in the first `Source` or a heavy computation in the second better use `concatLazy`. Here we start ticking from the very beginning. Not waiting for the first source to be consumed. But maybe you thought of that already.